### ISSUE TITLE
Method type parameter shared in chain expression

### ISSUE THREAD

**Issue by bannmann** [tomokinakamaru/silverchain#68]
Generic methods as introduced by #39 work fine, but I just noticed that they are shared with the chain (i.e. have a generic return value), which bloats Javadoc and potentially increases the number of states.

### Test AG file

```java
import com.example.FooBuilder;
import com.example.Foo;

import java.util.function.Function;

FooBuilder
{
    Foo
    addConverter<T>(Class<T> targetClass, Function<T, String> converter)*
    build();
}
```

### Expected result

```java
interface IFooBuilder {

  <T> IFooBuilder addConverter(Class<T> targetClass, java.util.function.Function<T, String> converter);

  com.example.Foo build();
}
```

### Actual result

```java
interface IFooBuilder {

  <T> state1.FooBuilder<T> addConverter(Class<T> targetClass, java.util.function.Function<T, String> converter);

  com.example.Foo build();
}
```

**Comment by tomokinakamaru** [comment]
I fixed the bug in PR #70!

@bannmann, can you check that the bug is correctly fixed? I checked it (of course), but it's better to double-check it.

**Comment by bannmann** [comment]
Yes, it fixes the bug, with spectacular results: PureTemplate's `GroupLoader` went down from 118 state classes to 10.

Thanks a lot!
